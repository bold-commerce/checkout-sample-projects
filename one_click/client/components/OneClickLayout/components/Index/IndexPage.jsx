import React, { useCallback, useContext, useEffect, useState }  from 'react';
import Card from '../Card';
import classNames from 'classnames';
import { Price } from '@boldcommerce/stacks-ui';
import { LineItems } from '../LineItems';
import { useCheckoutStore, useShippingAddress } from '@boldcommerce/checkout-react-components';
import { LoadingState } from '../LoadingState';
import { Header } from '../Header';
import { AppContext } from '../../context/AppContext';
import { PaymentMethod } from '../Payment';
import { CheckoutButton } from '../CheckoutButton';
import { useAnalytics } from '../../../../hooks';

const IndexPage = ({ onSectionChange, show }, ref) => {
  const { state } = useCheckoutStore();
  const { order_total, customer, addresses, shipping } = state.applicationState;
  const shippingAddressArray = [ addresses.shipping.address_line_1, addresses.shipping.address_line_2, addresses.shipping.city, addresses.shipping.province, addresses.shipping.postal_code, addresses.shipping.country ]
  const billingAddressArray = [ addresses.billing.address_line_1, addresses.billing.address_line_2, addresses.billing.city, addresses.billing.province, addresses.billing.postal_code, addresses.billing.country ];
  const shippingCombined = shippingAddressArray.filter((addressString) => addressString !== '').join(', ');
  const billingCombined = billingAddressArray.filter((addressString) => addressString !== '').join(', ');
  const { submitShippingAddress } = useShippingAddress();
  const [loading, setLoading] = useState(false);
  const { websiteName } = useContext(AppContext);
  const trackEvent = useAnalytics();

  const setDefaultAddress = useCallback(async () => {
    setLoading(true);
    try {
      await submitShippingAddress(customer.saved_addresses[0]);
    } catch(e) {
      setLoading(false);
    }
    setLoading(false);
  }, [customer.saved_addresses]);

  useEffect(() => {
    //if customer hasn't set a shipping address yet and they have a saved shipping address, set the shipping address to the first one. 
    if(addresses.shipping.length == 0 && customer.saved_addresses.length > 0){
      setDefaultAddress();
    }

    trackEvent('landing_page');
  }, []);

  return (
    <div ref={ref} className={classNames('Sidebar IndexPage', show ? 'Sidebar--Show' : 'IndexPage--Hide')}>
      <Header title={websiteName}/>
      <LineItems />
      <Card
        title={"Summary"}
        handleClick={() => onSectionChange('summary')}
        component={"/summary"}
        overview={<Price amount={order_total} />}
      />
      { customer?.email_address && (
        <Card
          description={customer.email_address}
          action={{label: "Not you?"}}
        />
      )}
      <Card
        title={"Shipping"}
        type={"shippingCard"}
        handleClick={() => onSectionChange('shipping')}
        description={loading ? '' : addresses.shipping.first_name ? `${addresses.shipping.first_name} ${addresses.shipping.last_name}` : 'No shipping address selected'}
      >
      { (loading && <LoadingState/>) || 
      ( 
        shipping.selected_shipping &&
        <>
          <div>{shippingCombined}</div>
          <div className="card-shipping-content">{shipping.selected_shipping.description} - <Price amount={shipping.selected_shipping.amount} /></div>
        </>
      )}
      </Card>
      <Card
        title={"Payment"}
        type={"paymentCard"}
      >
      {
        state.orderInfo.billingSameAsShipping ? 
        <div>Billing address same as shipping address</div> :
        <div>{billingCombined}</div>
      }
      </Card>
      <PaymentMethod />
      <CheckoutButton className="CheckoutButton CheckoutButton__Mobile" />
    </div>
  )
};

const IndexPageForwardedRef = React.forwardRef(IndexPage);

export default IndexPageForwardedRef;
