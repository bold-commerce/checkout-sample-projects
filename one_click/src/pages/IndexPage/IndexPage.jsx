import React, { useCallback, useContext, useEffect, useState }  from 'react';
import Card from '../../components/Card';
import classNames from 'classnames';
import { Price } from '@boldcommerce/stacks-ui';
import { LineItems } from '../../components/LineItems';
import { useCheckoutStore, useShippingAddress } from '@boldcommerce/checkout-react-components';
import { LoadingState } from '../../components/LoadingState';
import { Header } from '../../components/Header';
import { AppContext } from '../../context/AppContext';
import { PaymentMethod } from '../../components/Payment';
import { CheckoutButton } from '../../components/CheckoutButton';
import { useAnalytics } from '../../hooks';
import { useTranslation } from 'react-i18next';
import { SummaryCondensed } from '../../components/Summary';

const IndexPage = ({ onSectionChange, show }, ref) => {
  const { state } = useCheckoutStore();
  const { order_total, customer, addresses, shipping } = state.applicationState;
  const shippingAddressArray = [ addresses.shipping.address_line_1, addresses.shipping.address_line_2, addresses.shipping.city, addresses.shipping.province, addresses.shipping.postal_code, addresses.shipping.country ]
  const billingAddressArray = [ addresses.billing.address_line_1, addresses.billing.address_line_2, addresses.billing.city, addresses.billing.province, addresses.billing.postal_code, addresses.billing.country ];
  const shippingCombined = shippingAddressArray.filter((addressString) => addressString !== '').join(', ');
  const billingCombined = billingAddressArray.filter((addressString) => addressString !== '').join(', ');
  const { submitShippingAddress } = useShippingAddress();
  const [loading, setLoading] = useState(false);
  const { websiteName } = useContext(AppContext);
  const trackEvent = useAnalytics();
  const { t } = useTranslation();

  const setDefaultAddress = useCallback(async () => {
    setLoading(true);
    try {
      await submitShippingAddress(customer.saved_addresses[0]);
    } catch(e) {
      setLoading(false);
    }
    setLoading(false);
  }, [customer.saved_addresses]);

  useEffect(() => {
    //if customer hasn't set a shipping address yet and they have a saved shipping address, set the shipping address to the first one. 
    if(addresses.shipping.length == 0 && customer.saved_addresses.length > 0){
      setDefaultAddress();
    }

    trackEvent('landing_page');
  }, []);

  return (
    <div ref={ref} className={classNames('Sidebar IndexPage', show ? 'Sidebar--Show' : 'IndexPage--Hide')}>
      <Header title={websiteName}/>
      <LineItems />
      <SummaryCondensed handleClick={() => onSectionChange('summary')}/>
      { customer?.email_address && (
        <Card
          description={customer.email_address}
          action={{
            label: t('customer.not_you'),
            onClick: () => { window.location = process.env.LOGOUT_URL }
          }}
        />
      )}
      <Card
        title={t('shipping.title')}
        type={"shippingCard"}
        handleClick={() => onSectionChange('shipping')}
        description={loading ? '' : addresses.shipping.first_name ? `${addresses.shipping.first_name} ${addresses.shipping.last_name}` : t('shipping.empty')}
      >
      { (loading && <LoadingState/>) || 
      ( 
        shipping.selected_shipping &&
        <>
          <div>{shippingCombined}</div>
          <div className="card-shipping-content">{shipping.selected_shipping.description} - <Price amount={shipping.selected_shipping.amount} moneyFormatString={t('currency_format')}/></div>
        </>
      )}
      </Card>
      <Card
        title={t('payment.title')}
        type={"paymentCard"}
      >
      {
        state.orderInfo.billingSameAsShipping ? 
        <div>{t('payment.billing_same_as_shipping')}</div> :
        <div>{billingCombined}</div>
      }
      </Card>
      <PaymentMethod />
      <CheckoutButton className="CheckoutButton CheckoutButton__Mobile" />
    </div>
  )
};

const IndexPageForwardedRef = React.forwardRef(IndexPage);

export default IndexPageForwardedRef;
