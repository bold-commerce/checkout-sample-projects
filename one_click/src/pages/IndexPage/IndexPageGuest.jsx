import React, { useContext, useState } from "react";
import { Button } from "@boldcommerce/stacks-ui";
import { useShippingAddress, useCountryInfo, useCheckoutStore, useCustomer } from "@boldcommerce/checkout-react-components";
import { ShippingLines } from "../../components/ShippingLines";
import { AppContext } from "../../context/AppContext";
import { Address } from "../../components/Address";
import { Header } from "../../components/Header";
import { Link } from "react-router-dom";
import classNames from "classnames";
import { CustomerInfo } from "../../components/CustomerInfo";
import { SummaryCondensed } from "../../components/Summary";
import { useTranslation } from 'react-i18next';

const IndexPageGuest = ({ onSectionChange, show }, ref) => {
  const { websiteName } = useContext(AppContext);
  const { state } = useCheckoutStore();
  const { data: shippingAddress, errors: shippingErrors, submitShippingAddress } = useShippingAddress();
  const [ address, setAddress ] = useState(shippingAddress);
  const { data: countryInfo } = useCountryInfo(address);
  const { data: customer, errors: customerErrors } = useCustomer();
  const { t } = useTranslation();

  const requiredAddressFields = ['first_name','last_name','address_line_1','city'];
  let provincePlaceholder = countryInfo.provinceLabel;

  return (
    <div ref={ref} className={classNames('Sidebar IndexGuest', show ? 'Sidebar--Show' : 'IndexPage--Hide')}>
      <Header title={websiteName} />
      <SummaryCondensed onSectionChange={() => onSectionChange('summary')}/>
      <CustomerInfo />
      <div className="IndexGuest__ShippingAddress">
        <h2 className="IndexGuest__Title">{t('shipping.address')}</h2>
        <Address
          address={address}
          onChange={(d) => setAddress((prevAddress) => ({
            ...prevAddress,
            ...d,
          }))}
          errors={shippingErrors}
          countries={countryInfo.countries}
          provinces={countryInfo.provinces}
          showPostalCode={countryInfo.showPostalCode}
          showProvince={countryInfo.showProvince}
          submit={() => submitShippingAddress(address)}
          provinceLabel={provincePlaceholder}
          requiredAddressFields={requiredAddressFields}
        />
      </div>
      <div className="IndexGuest__ShippingMethod">
        <ShippingLines activePage={show} disabled={Boolean(shippingErrors)} />
      </div>
      <div className="IndexGuest__Footer">
        <Button
          className="IndexGuest__Billing-btn"
          onClick={() => onSectionChange('billing')}
          disabled={!(customer.email_address && state.applicationState.shipping.selected_shipping && !shippingErrors && !customerErrors)}
          primary
        >
          {t('continue_to_billing')}
        </Button>
        <div className="IndexGuest__Footer--login">
          <Link to="#">{t('back_to_login')}</Link>
        </div>          
        <div className="IndexGuest__Rights">{`${t('all_rights_reserved')} ${websiteName}`}</div>
      </div>
    </div>
  )
}

const IndexPageGuestForwardedRef = React.forwardRef(IndexPageGuest);

export default IndexPageGuestForwardedRef;