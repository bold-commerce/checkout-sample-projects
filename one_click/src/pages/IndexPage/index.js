export { default as IndexPage } from "./IndexPage";
export { default as IndexPageGuest } from "./IndexPageGuest";