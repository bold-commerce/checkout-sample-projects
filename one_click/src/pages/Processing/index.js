export { default as ProcessingOrder } from './Processing'