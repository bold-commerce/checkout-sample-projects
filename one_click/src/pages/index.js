export * from './Confirmation';
export * from './IndexPage';
export * from './Processing';
export * from './Inventory';