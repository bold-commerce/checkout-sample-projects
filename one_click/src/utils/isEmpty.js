const isEmpty = (obj) => {
  return JSON.stringify(obj) === JSON.stringify({});
};

export default isEmpty;