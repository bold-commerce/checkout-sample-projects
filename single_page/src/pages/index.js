export * from './ConfirmationPage';
export * from './IndexPage';
export * from './ProcessingPage';
