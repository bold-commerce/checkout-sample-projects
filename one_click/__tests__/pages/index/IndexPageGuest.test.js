import React from 'react';
import { IndexPageGuest } from '../../../src/pages/IndexPage';
import { render } from '@testing-library/react';
import { exampleLineItems as MOCKexampleLineItems } from '../../utils/lineItemHelpers';
import { testApplicationState as MOCKexampleApplicationState } from '../../utils/applicationStateHelper';
import {
    countries as MOCKcountries,
    caProvinces as MOCKprovinces
} from '../../utils/addressHelpers';
import { exampleShippingLines as MOCKexampleShippingLines } from '../../utils/shippingLinesHelper';
import { BrowserRouter } from 'react-router-dom';

jest.mock('@boldcommerce/checkout-react-components', () => ({
    ...jest.requireActual('@boldcommerce/checkout-react-components'),
    useLineItems: () => ({
        data: MOCKexampleLineItems,
        updateLineItemQuantity: (() => {}),
        removeLineItem: (() => {})        
    }),
    useCheckoutStore: () => ({
        state: {
            orderInfo: {
                billingSameAsShipping:true,
                orderStatus: ""
            },
            errors: { order: null },
            loadingStatus: { isLoading: false },
            applicationState: MOCKexampleApplicationState
        }
    }),
    useCustomer: () => ({
        data: {
            platform_id: 123
        }
    }),
    usePaymentIframe: () => ({
        data: {
            url: 'test.url',
            loadingStatus: 'fulfilled',
            paymentIframeOnLoaded: (() => {})
        }
    }),
    useShippingLines: () => ({
        data: MOCKexampleShippingLines
    }),
    useShippingAddress: () => ({
        data: {},
        submitShippingAddress: (() => {})
    }),
    useBillingAddress: () => ({
        data: {}
    }),
    useDiscount: () => ({
        data: '?',
        applyDiscount: (() => {})
    }),
    useBillingSameAsShipping: () => ({
        data: true,
        setBillingSameAsShipping: (() => {})
    }),
    useCountryInfo: () => ({
        data: {
            countries: MOCKcountries,
            provinceLabel: "province",
            provinces: MOCKprovinces,
            showPostalCode: true,
            showProvince: true
        }
    }),
})).mock('react', () => ({
    ...jest.requireActual('react'),
    useContext: () => ({
        websiteName: 'TestSite'
    })
})).mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useLocation: () => ({
        pathname: '/' 
    })
}));

describe('IndexPageGuest', () => {
  test('renders IndexPageGuest component', () => {
    const { asFragment } = render(<BrowserRouter><IndexPageGuest show /></BrowserRouter> )
    expect(asFragment()).toMatchSnapshot();
  })

  test('renders IndexPageGuest component', () => {
    const { asFragment } = render(<BrowserRouter><IndexPageGuest show={false} /></BrowserRouter> )
    expect(asFragment()).toMatchSnapshot();
  })
})