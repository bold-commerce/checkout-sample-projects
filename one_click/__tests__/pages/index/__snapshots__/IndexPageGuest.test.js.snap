// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPageGuest renders IndexPageGuest component 1`] = `
<DocumentFragment>
  <div
    class="Sidebar IndexGuest Sidebar--Show"
  >
    <div
      class="Header"
    >
      <h1
        class="Header__Title"
      >
        TestSite
      </h1>
      <button
        aria-label="Close"
        class="Header__Times"
      >
        <svg
          aria-labelledby="button"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            opacity="0.8"
          >
            <path
              clip-rule="evenodd"
              d="M17.7071 7.70706L7.70712 17.7071L6.29291 16.2928L16.2929 6.29285L17.7071 7.70706Z"
              fill="black"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M16.2929 17.7071L6.29291 7.70706L7.70712 6.29285L17.7071 16.2928L16.2929 17.7071Z"
              fill="black"
              fill-rule="evenodd"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="IndexGuest__Summary"
    >
      <button
        class="IndexGuest__Summary__Btn"
      >
        <svg
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.5858 12L6.29289 3.70711C5.90237 3.31658 5.90237 2.68342 6.29289 2.29289C6.68342 1.90237 7.31658 1.90237 7.70711 2.29289L16.7071 11.2929C17.0976 11.6834 17.0976 12.3166 16.7071 12.7071L7.70711 21.7071C7.31658 22.0976 6.68342 22.0976 6.29289 21.7071C5.90237 21.3166 5.90237 20.6834 6.29289 20.2929L14.5858 12Z"
            fill="black"
          />
        </svg>
        <h2>
          summary.title
        </h2>
        <div
          class="stx-price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            $262.00
          </p>
        </div>
      </button>
    </div>
    <div
      class="IndexGuest__Customer"
    >
      <h2
        class="IndexGuest__Title"
      >
        customer.info
      </h2>
      <div
        class="IndexGuest__Login"
      >
        customer.already_have_account
        <a
          href="/"
        >
          customer.login
        </a>
      </div>
      <div
        class="stx-field stx-field--with-input"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField1"
            name="email_address"
            placeholder="customer.email"
            type="email"
          />
        </div>
      </div>
      <div
        class="stx-field stx-checkbox-field__field"
      >
        <div
          class="stx-checkbox-field__div"
        >
          <span
            class="stx-checkbox stx-checkbox-field__checkbox"
          >
            <input
              class="stx-checkbox__input"
              id="stxField2"
              type="checkbox"
            />
            <span
              class="stx-checkbox__mount"
            />
          </span>
          <label
            class="stx-checkbox-field__label"
            for="stxField2"
          >
            customer.subscribe
          </label>
        </div>
      </div>
    </div>
    <div
      class="IndexGuest__ShippingAddress"
    >
      <h2
        class="IndexGuest__Title"
      >
        shipping.address
      </h2>
      <div
        class="FieldSet--Address"
      >
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--FirstName"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField3"
                name="first_name"
                placeholder="address.first_name"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="stx-field stx-field--with-input Field Field--LastName"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField4"
                name="last_name"
                placeholder="address.last_name"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Company"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField5"
                name="business_name"
                placeholder="address.business_name_optional"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Address"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField6"
                name="address_line_1"
                placeholder="address.address_line_1"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="stx-field stx-field--with-input Field Field--Address2"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField7"
                name="address_line_2"
                placeholder="address.address_line_2"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--City"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField8"
                name="city"
                placeholder="address.city"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup FieldGroup__Country"
        >
          <div
            class="stx-field stx-field--with-select SelectField Field--Country"
          >
            <div
              class="stx-select stx-select--with-placeholder"
            >
              <select
                class="stx-select__select-element"
                id="stxField9"
              >
                <option
                  hidden=""
                  value="___PLACEHOLDER___"
                >
                  address.country_hint
                </option>
                <option
                  value="CA"
                >
                  Canada
                </option>
                <option
                  value="US"
                >
                  United States
                </option>
              </select>
              <span
                class="stx-select__icon"
              >
                <svg
                  class="stx-select-icon__icon"
                  viewBox="0 0 24 24"
                >
                  <path
                    class="stx-select-icon__path"
                    d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                    fill="#6B6B6B"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Phone"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField10"
                name="phone_number"
                placeholder="address.phone_number_optional"
                type="tel"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="IndexGuest__ShippingMethod"
    >
      <section
        class="FieldSet FieldSet--ShippingMethod"
      >
        <div
          class="FieldSet__Header"
        >
          <h3
            class="FieldSet__Heading"
          >
            shipping.method
          </h3>
        </div>
        <div
          class="FieldSet__Content"
        />
      </section>
    </div>
    <div
      class="IndexGuest__Footer"
    >
      <button
        aria-busy="false"
        aria-disabled="true"
        class="stx-button stx-button--primary stx-button--is-disabled IndexGuest__Billing-btn"
        disabled=""
      >
        <span
          class="stx-button__text"
        >
          continue_to_billing
        </span>
        <span
          class="stx-loading-spinner stx-button__spinner"
        />
      </button>
      <div
        class="IndexGuest__Footer--login"
      >
        <a
          href="/"
        >
          back_to_login
        </a>
      </div>
      <div
        class="IndexGuest__Rights"
      >
        all_rights_reserved TestSite
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`IndexPageGuest renders IndexPageGuest component 2`] = `
<DocumentFragment>
  <div
    class="Sidebar IndexGuest IndexPage--Hide"
  >
    <div
      class="Header"
    >
      <h1
        class="Header__Title"
      >
        TestSite
      </h1>
      <button
        aria-label="Close"
        class="Header__Times"
      >
        <svg
          aria-labelledby="button"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            opacity="0.8"
          >
            <path
              clip-rule="evenodd"
              d="M17.7071 7.70706L7.70712 17.7071L6.29291 16.2928L16.2929 6.29285L17.7071 7.70706Z"
              fill="black"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M16.2929 17.7071L6.29291 7.70706L7.70712 6.29285L17.7071 16.2928L16.2929 17.7071Z"
              fill="black"
              fill-rule="evenodd"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="IndexGuest__Summary"
    >
      <button
        class="IndexGuest__Summary__Btn"
      >
        <svg
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.5858 12L6.29289 3.70711C5.90237 3.31658 5.90237 2.68342 6.29289 2.29289C6.68342 1.90237 7.31658 1.90237 7.70711 2.29289L16.7071 11.2929C17.0976 11.6834 17.0976 12.3166 16.7071 12.7071L7.70711 21.7071C7.31658 22.0976 6.68342 22.0976 6.29289 21.7071C5.90237 21.3166 5.90237 20.6834 6.29289 20.2929L14.5858 12Z"
            fill="black"
          />
        </svg>
        <h2>
          summary.title
        </h2>
        <div
          class="stx-price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            $262.00
          </p>
        </div>
      </button>
    </div>
    <div
      class="IndexGuest__Customer"
    >
      <h2
        class="IndexGuest__Title"
      >
        customer.info
      </h2>
      <div
        class="IndexGuest__Login"
      >
        customer.already_have_account
        <a
          href="/"
        >
          customer.login
        </a>
      </div>
      <div
        class="stx-field stx-field--with-input"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField11"
            name="email_address"
            placeholder="customer.email"
            type="email"
          />
        </div>
      </div>
      <div
        class="stx-field stx-checkbox-field__field"
      >
        <div
          class="stx-checkbox-field__div"
        >
          <span
            class="stx-checkbox stx-checkbox-field__checkbox"
          >
            <input
              class="stx-checkbox__input"
              id="stxField12"
              type="checkbox"
            />
            <span
              class="stx-checkbox__mount"
            />
          </span>
          <label
            class="stx-checkbox-field__label"
            for="stxField12"
          >
            customer.subscribe
          </label>
        </div>
      </div>
    </div>
    <div
      class="IndexGuest__ShippingAddress"
    >
      <h2
        class="IndexGuest__Title"
      >
        shipping.address
      </h2>
      <div
        class="FieldSet--Address"
      >
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--FirstName"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField13"
                name="first_name"
                placeholder="address.first_name"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="stx-field stx-field--with-input Field Field--LastName"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField14"
                name="last_name"
                placeholder="address.last_name"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Company"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField15"
                name="business_name"
                placeholder="address.business_name_optional"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Address"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField16"
                name="address_line_1"
                placeholder="address.address_line_1"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="stx-field stx-field--with-input Field Field--Address2"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField17"
                name="address_line_2"
                placeholder="address.address_line_2"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--City"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField18"
                name="city"
                placeholder="address.city"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup FieldGroup__Country"
        >
          <div
            class="stx-field stx-field--with-select SelectField Field--Country"
          >
            <div
              class="stx-select stx-select--with-placeholder"
            >
              <select
                class="stx-select__select-element"
                id="stxField19"
              >
                <option
                  hidden=""
                  value="___PLACEHOLDER___"
                >
                  address.country_hint
                </option>
                <option
                  value="CA"
                >
                  Canada
                </option>
                <option
                  value="US"
                >
                  United States
                </option>
              </select>
              <span
                class="stx-select__icon"
              >
                <svg
                  class="stx-select-icon__icon"
                  viewBox="0 0 24 24"
                >
                  <path
                    class="stx-select-icon__path"
                    d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                    fill="#6B6B6B"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Phone"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField20"
                name="phone_number"
                placeholder="address.phone_number_optional"
                type="tel"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="IndexGuest__ShippingMethod"
    >
      <section
        class="FieldSet FieldSet--ShippingMethod"
      >
        <div
          class="FieldSet__Header"
        >
          <h3
            class="FieldSet__Heading"
          >
            shipping.method
          </h3>
        </div>
        <div
          class="FieldSet__Content"
        >
          <p>
            shipping.options_description
          </p>
        </div>
      </section>
    </div>
    <div
      class="IndexGuest__Footer"
    >
      <button
        aria-busy="false"
        aria-disabled="true"
        class="stx-button stx-button--primary stx-button--is-disabled IndexGuest__Billing-btn"
        disabled=""
      >
        <span
          class="stx-button__text"
        >
          continue_to_billing
        </span>
        <span
          class="stx-loading-spinner stx-button__spinner"
        />
      </button>
      <div
        class="IndexGuest__Footer--login"
      >
        <a
          href="/"
        >
          back_to_login
        </a>
      </div>
      <div
        class="IndexGuest__Rights"
      >
        all_rights_reserved TestSite
      </div>
    </div>
  </div>
</DocumentFragment>
`;
